pkgs/tools/compression/pbzip2/default.nix

```diff
@@ -11,9 +11,15 @@ stdenv.mkDerivation rec {
     sha256 = "1i7rql77ac33lz7lzrjyl9b16mqizmdkb8hv295a493f7vh1nhmx";
   };
 
-  buildInputs = [ bzip2 ];
+  buildInputs = [ bzip2 ] ++ 
+    stdenv.lib.optional stdenv.isDarwin [ clang ];
+
+  preBuild = stdenv.lib.optionalString stdenv.isDarwin ''
+    substituteInPlace Makefile --replace g++ clang++
```

Comment:
Couldn't find anything in the stdenv setup to indicate how to determine stdenv isn't built on gcc. Just switched it to replace to c++ and things seem to still work at least. So no need for this to be clang/osx specific.